app/scripts/controllers/newfromtemplate.js

```diff
@@ -355,6 +367,15 @@ angular.module('openshiftConsole')
             $scope.parameterDisplayNames[parameter.name] = parameter.displayName || parameter.name;
           });
 
+          if($routeParams.templateParamsMap) {
+            var templateParams = getValidTemplateParamsMap();
+            _.each($scope.template.parameters, function(parameter) {
+              if (templateParams[parameter.name]) {
+                parameter.value = templateParams[parameter.name];
```

Comment:
Did we switch back to array for this controller? I think it would be better to use a map everywhere to be consistent if possible.